### DISCUSSION TITLE
How can I get my scenarios tag values in a before hook?

### DISCUSSION THREAD

**Discussion by henriqueChoreograph** [badeball/cypress-cucumber-preprocessor#1157] [Q&A] [answered]
I have some scenarios to validate some specific bugs opened in Jira and these scenarios are tagged with the bug Id so I can keep track of it.

What I want now is to get this tag value in a beforeEach hook and conditionally run my test.

Example:

```
Feature: Testing bugs
  Here I just have scenarios to test bugs

    @jiraId_001
    Scenario: Testing bug 1
  
    @jiraId_002
    Scenario: Testing bug 2
```

I want to get the "jiraId_001" tag value and skip the test if the string value matches my criteria.

With this string value, I can check the item status in Jira and skip the test if the defect is still open.

Does anyone know how to get the scenario tag values?

**Comment by badeball** [comment, marked as answer]
In cucumber-js, you can skip a test from `Before` hooks, like shown below. This hasn't been implemented here though, but can be tracked at https://github.com/badeball/cypress-cucumber-preprocessor/issues/1159.

```js
Before(({ pickle }) => {
  if (pickle.tags.some(tag => tag.name === "@foo")) {
    return "skip";
  }
});
```

**Reply by badeball** [reply]
This has been implemented as of v20.0.2.

**Comment by Benjamin-Re** [comment]
Where do I place the Before hook? I tried to add it as a global hook in index.ts but it causes this error:

`Error: The following error originated from your test code, not from Cypress.

  > Expected to find a global registry (this usually means you are trying to define steps or hooks in support/e2e.js, which is not supported) (this might be a bug, please report at https://github.com/badeball/cypress-cucumber-preprocessor)`
